### PR TITLE
InternalCalibration: better debug messages for R scripts

### LINKED ISSUE #2251: InternalCalibration: Problems with R scripts
I'm having some trouble with the functionality for generating quality control plots using R scripts in InternalCalibration.

First, I tried running it on a server that had the "Rscript" executable, which however turned out to be broken (due to a missing library). The log messages in that case were confusing:
```
Could not find 'Rscript' executable. Make sure it's in your system path and try again.

Calibration failed. See error message above!
Running R script '/nfs/t17_project/OpenMS/share/OpenMS/SCRIPTS/InternalCalibration_Models.R' ...
```
The first problem is that the "Running R script" message should not appear if "Rscript" didn't work. The second problem is that the output of trying to run "Rscript" is not available, which makes it hard to diagnose what the problem is without digging through the OpenMS code.
The third (minor) problem is that the calibration actually worked, only generating the quality control plot didn't, so "Calibration failed" is not really true.

I then moved to a server with a functioning "Rscript", where the first quality control plotting script ran through successfully and generated the "models" plot:
`Running R script '/nfs/t17_project/OpenMS/share/OpenMS/SCRIPTS/InternalCalibration_Models.R' ... success`
However, after that the process stalled, there was no further log output, and I stopped it after about an hour. A PNG file for the "residuals" plot was created, but was completely white.

### PR DESCRIPTION
**Pull request by cbielow**
fixes #2251

[FIX] InternalCalibration: much better error descriptions if Rscript executable is not found or broken.
[FIX] InternalCalibration: R script to generate ppm-error plots now works properly with large peptide ID data

### CODE REVIEW

**Comment by cbielow** [comment]
tests pass, but lets wait with merging until I confirmed one last thing on Hendriks data

**Comment by cbielow** [comment]
ok, PR is ready

**Comment by timosachsenberg** [comment]
would it make sense to also open this PR on master?

File: src/openms/source/SYSTEM/RWrapper.cpp [hunk 1]
```diff
@@ -46,62 +46,97 @@
...
     }
+    if (verbose) LOG_INFO << " success" << std::endl;
+
+    if (verbose) LOG_INFO << "Trying to invoke 'Rscript' ...";
+    if (p.exitStatus() != QProcess::NormalExit || p.exitCode() != 0)
+    {
+      if (verbose)
+      {
+        LOG_INFO << " failed" << std::endl;
+        LOG_ERROR << "Error: 'Rscript' executable returned with error (command: 'Rscript " << args.join(" ").toStdString() << "'\n"
```
Comment (hendrikweisser) [inline src/openms/source/SYSTEM/RWrapper.cpp]: Missing ")" to pair with "(command: ...".

File: src/openms/source/SYSTEM/RWrapper.cpp [hunk 2]
```diff
@@ -46,62 +46,97 @@
...
 
-    LOG_INFO << "Running R script '" << fullscript << "' ..." << std::flush;
-    if (!findR())
-    {
-      return false;
-    }
-
-    QProcess p;
-    p.setProcessChannelMode(QProcess::MergedChannels);
+    if (verbose) LOG_INFO << "Running R script '" << fullscript << "' ...";
```
Comment (hendrikweisser) [inline src/openms/source/SYSTEM/RWrapper.cpp]: I'm generally not a fan of the space before "..." (here and elsewhere), but that may be a matter of (typographic) taste.

**Comment by timosachsenberg** [comment]
@hendrikweisser sorry for the quick merge (I wanted to sync master with develop). I can add these changes on top.

**Comment by timosachsenberg** [comment]
I put it on the merge branch https://github.com/OpenMS/OpenMS/pull/2260

**Comment by hendrikweisser** [comment]
@timosachsenberg: Cool, thanks!

**Comment by cbielow** [comment]
yes

Am 10/25/2016 um 10:56 AM schrieb Timo Sachsenberg:

**Pull request merged**